### ISSUE TITLE
Could not copy ASP.NET Core dll, the file is locked by IIS Worker Process

### ISSUE THREAD

**Issue by justintoth** [aspnet/AspNetCore#10088]
### Describe the bug
I'm attempting to develop an ASP.NET Core web application running locally under IIS on Windows 10. Every time that I run the app then try to rebuild the ASP.NET project, it errors:

> 
> Error	MSB3027	Could not copy "obj\Debug\netcoreapp2.2\Housters.Web.dll" to "bin\Debug\Housters.Web.dll". Exceeded retry count of 10. Failed. The file is locked by: "IIS Worker Process (15496)"	Web	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets	4203	
> Error	MSB3021	Unable to copy file "obj\Debug\netcoreapp2.2\Housters.Web.dll" to "bin\Debug\Housters.Web.dll". The process cannot access the file 'bin\Debug\Housters.Web.dll' because it is being used by another process.	Web	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets	4203

	
In order to work around the issue, I need to recycle the IIS app pool *every single time* before rebuilding the solution. This is not a productive way to spend my time, so am looking for a resolution on this...

### To Reproduce
1. Using this version of ASP.NET Core '2.2'
2. Using this version of Visual Studio '2019'

### Expected behavior
The ASP.NET Core project can rebuild successfully when when it's running in IIS.

**Comment by brockallen** [comment]
I've been seeing this bug since VS 2017. // @Andrew-MSFT

**Comment by Andrew-MSFT** [comment]
Adding @BillHiebert

**Comment by brockallen** [comment]
This last week while teaching my workshop, I was seeing it about every 20% of the time when I launch my app under IIS express, alt-tab back to VS, make a quick change, Ctrl-F5, then DLLs locked error. I'm always using Ctrl-F5, and never F5.

**Comment by anurse** [comment]
I've actually seen this sometimes as well. @BillHiebert any thoughts? It seems like VS isn't terminating IIS Express in preparation for the build?

**Comment by BillHiebert** [comment]
VS should be terminating the process.  Is asp.net being hosting in-process or out-of-process? If out of process, it will restart IIS Express but that should not start dot.net.exe until an http request is received from the browser. When a request s received, VS should  be blocking it until the build completes to ensure the files aren't locked.

I suspect the IIS case, is because the application pool process is being started external to Visual Studio and VS is not tracking it properly when this occurs.  This issue is being tracked [here ](https://developercommunity.visualstudio.com/content/problem/546858/bin-files-locked-by-iis-worker-process-1.html)

**Comment by anurse** [comment]
@justintoth does the VS feedback item Bill linked cover your scenario? If so, we can close this issue since it's being tracked in VS.

**Comment by justintoth** [comment]
@anurse Yes it does, thanks.

**Comment by dnknitro** [comment]
Any updates on the issue? Same behavior here...

**Comment by davidfowl** [comment]
@dnknitro Can you open a new issue with your symptoms and version information?

**Comment by anurse** [comment]
@dnknitro If you're seeing the same issue, please let us know on the VS Feedback item that was linked above: https://developercommunity.visualstudio.com/content/problem/546858/bin-files-locked-by-iis-worker-process-1.html

This is an issue in VS, and the VS team is investigating it in that issue.

**Comment by dnknitro** [comment]
I have just posted my Output window Build log to that VS Feedback item.

**Comment by rockarts** [comment]
I can fix the issue by killing the IIS worker process. VS 2017 then asks if I want to restart VS under different permissions to debug. Seems like it could be related to trying to debug on IIS without elevated permissions.
